### ISSUE TITLE
parser: should not parse `as` or `satisfies` expression for js

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#6427]
Please check out the [Playground](https://playground.oxc.rs/#eNpFjtENgzAMRFdB/kYdIAOwh0lNlSqxkW3aIsTuTaigX76zT++8QYQA2KF1yCv0IBA20IXbsJUdPxBcF+ohJ/ZTW5SZLrOWUfLpXJFtEi0QJsxGew8zqpEeRFk00pAyMZYKACfz29Ogphr/SFV9Qc5FlDs9qH1VTUmcpvQ/savkIcu7VbxIR7HK/rXv+xefgU/r) 
```ts
a as any
```
The above code should throw an error if TS doesn't enable, because `as` is not a keyword in JavaScript

**Comment by Boshen** [comment]
Need a way to check this systematically 🤔

**Comment by Boshen** [comment]
Found it: `getJSSyntacticDiagnosticsForFile` in TypeScript/src/compiler/program.ts
